### PR TITLE
Add BSD Support

### PR DESCRIPTION
**Pull request by jabbate19**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Adds cfg macros for freeBSD support. Only thing not supported is netstat

#### Which issue(s) this PR fixes:
N/A

### CODE REVIEW

File: implants/lib/eldritch/Cargo.toml [hunk 1]
```diff
@@ -70,6 +69,9 @@ winreg = { workspace = true }
 [target.'cfg(not(windows))'.dependencies]
 pnet = { workspace = true }
 
+[target.'cfg(not(target_os = "freebsd"))'.dependencies]
```
Comment (hulto) [inline implants/lib/eldritch/Cargo.toml:72]: Do we want to support all the BSDs?
- "freebsd"
- "openbsd"
- "netbsd"

**Review by hulto — APPROVED** [review]

**Review by hulto — CHANGES_REQUESTED** [review]
Ope wrong button

File: implants/lib/eldritch/Cargo.toml (hunk 1 above)
Comment (jabbate19) [reply implants/lib/eldritch/Cargo.toml:72]: I can check if my changes work on the other BSDs, but not sure if it will

Comment (hulto) [reply implants/lib/eldritch/Cargo.toml:72]: Yeah lmk how it goes.
If it doesn't we should add them to the exclude otherwise what a W!

**Comment by hulto** [comment]
On further investigation i don't think we should support netbsd.
image

File: implants/lib/eldritch/Cargo.toml (hunk 1 above)
Comment (hulto) [reply implants/lib/eldritch/Cargo.toml:78]: We will only be supporting freebsd at this time.

**Review by hulto — APPROVED** [review]

**Pull request merged**